src/embedded.js

```diff
@@ -606,12 +608,23 @@
                 window.addEventListener('scroll', this.fixIframe);
             }
 
+            // Close the iframe if page fails to initialize within 15 seconds
+            setTimeout(function() {
+              if (HelloSign.isSignerPageInitialized === false) {
+                HelloSign.reportError('Signer page failed to initialize within 15 seconds.', document.location.href);
+                HelloSign.close();
+              }
+            }, HelloSign.DEFAULT_HEALTH_CHECK_TIME);
```

Comment:
Also, store the timeout handle instead of a flag and just call `clearTimeout` when you receive the 'initialize' event.